smac/optimizer/base_prior.py

```diff
@@ -0,0 +1,395 @@
+
+import numpy as np
+import scipy.stats as sps
+
+
+class BasePrior(object):
```

Comment:
could we call this file gp_base_priors?

Reply:
And move this file to the epm directory?

Reply:
And also add type hints to all GP files?